barretenberg/cpp/src/barretenberg/vm/tests/AvmMini_execution.test.cpp

```diff
@@ -406,23 +403,19 @@ TEST_F(AvmMiniExecutionTests, jumpAndCalldatacopy)
     auto bytecode = hex_to_bytes(bytecode_hex);
     auto instructions = Deserialization::parse(bytecode);
 
-    EXPECT_EQ(instructions.size(), 5);
+    EXPECT_THAT(instructions, SizeIs(5));
 
     // We test parsing steps for CALLDATACOPY and JUMP.
 
     // CALLDATACOPY
     EXPECT_EQ(instructions.at(0).op_code, OpCode::CALLDATACOPY);
-    EXPECT_EQ(instructions.at(0).operands.size(), 3);
 
     auto operands = instructions.at(0).operands;
-    EXPECT_EQ(std::get<uint32_t>(operands.at(0)), 0);
-    EXPECT_EQ(std::get<uint32_t>(operands.at(1)), 2);
-    EXPECT_EQ(std::get<uint32_t>(operands.at(2)), 10);
+    EXPECT_THAT(operands, ElementsAre(VariantWith<uint32_t>(0), VariantWith<uint32_t>(2), VariantWith<uint32_t>(10)));
 
     // JUMP
     EXPECT_EQ(instructions.at(1).op_code, OpCode::JUMP);
-    EXPECT_EQ(instructions.at(1).operands.size(), 1);
-    EXPECT_EQ(std::get<uint32_t>(instructions.at(1).operands.at(0)), 3);
+    EXPECT_THAT(instructions.at(1).operands, ElementsAre(VariantWith<uint32_t>(3)));
 
     auto trace = Execution::gen_trace(instructions, std::vector<FF>{ 13, 156 });
 
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: line 425]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->You could try

```
EXPECT_THAT(trace, ElementsAre(
 _, // any
 Eq(Field(&classname::avmMini_pc), 1),
...
Eq(Field(&classname::avmMini_pc), 4),
));
```

This one is more optional since it's not _a lot_ more readable (it would be more readable if we could first map `.avmMini_pc`. The objective in general is that in tests there should be mostly no computations. The following lines are do have quite some.<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/4502?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->

Reply:
Yeah it is not more readable IMHO. I thought we could create a custom matcher to extract .avmMini_pc but just for this it is not justified.